karapace/typing.py

```diff
@@ -59,23 +59,55 @@ class Mode(StrEnum):
     readwrite = "READWRITE"
 
 
-class Version(int):
+class Version:
     LATEST_VERSION_TAG: ClassVar[str] = "latest"
     MINUS_1_VERSION_TAG: ClassVar[int] = -1
 
-    def __new__(cls, version: int) -> Version:
+    def __init__(self, version: int) -> None:
         if not isinstance(version, int):
             raise InvalidVersion(f"Invalid version {version}")
-        if (version < cls.MINUS_1_VERSION_TAG) or (version == 0):
+        if (version < Version.MINUS_1_VERSION_TAG) or (version == 0):
             raise InvalidVersion(f"Invalid version {version}")
-        return super().__new__(cls, version)
+        self._value = version
 
     def __str__(self) -> str:
-        return f"{int(self)}"
+        return f"{int(self._value)}"
 
     def __repr__(self) -> str:
-        return f"Version={int(self)}"
+        return f"Version({int(self._value)})"
+
+    def __gt__(self, other: Version) -> bool:
```

Comment:
Usually you do not need to implement all of these methods, you can just implement one of them and decorate the class with `@functools.total_ordering`.

Reply:
`@functools.total_ordering` comes with some performance overhead, docs advise to implement these functions if the performance hit is too much.